### PR TITLE
fix(toolbar-buttons) Hide rec and livestream buttons for non-moderators

### PR DESCRIPTION
**Pull request by horymury**

### CODE REVIEW

**Comment by damencho** [comment]
Is that the only place where isJwtFeatureEnabled( break the logic?

**Comment by horymury** [comment]
> Is that the only place where isJwtFeatureEnabled( break the logic?

Yes, it is also used for desktop share- but we allow desktop share for non-moderators, and apparently we allow sip invites as well for non-moderators

**Comment by damencho** [comment]
> sip invites as well for non-moderators

Nope, we don't allow sip outgoing calls for non moderators. Can you fix and that?

**Comment by horymury** [comment]
> > sip invites as well for non-moderators
> 
> Nope, we don't allow sip outgoing calls for non moderators. Can you fix and that?

done

**Review by saghul — DISMISSED** [review]

File: react/features/recording/components/LiveStream/AbstractLiveStreamButton.ts [hunk 1]
```diff
@@ -131,8 +131,9 @@ export function _mapStateToProps(state: IReduxState, ownProps: IProps) {
         const isModerator = isLocalParticipantModerator(state);
         const liveStreaming = getLiveStreaming(state);
 
-        visible = isModerator && liveStreaming.enabled;
-        visible = isJwtFeatureEnabled(state, 'livestreaming', visible);
+        visible = isModerator
+            ? isJwtFeatureEnabled(state, 'livestreaming', liveStreaming.enabled)
```
Comment (damencho) [inline react/features/recording/components/LiveStream/AbstractLiveStreamButton.ts:135]: But this means that if there is a jwt with the feature enabled, but the config has livestreaming disabled we will still show the button.

File: react/features/recording/functions.ts [hunk 2]
```diff
@@ -172,8 +172,9 @@ export function getRecordButtonProps(state: IReduxState) {
 
     const dropboxEnabled = isDropboxEnabled(state);
 
-    visible = isModerator && (recordingService?.enabled || localRecordingEnabled || dropboxEnabled);
-    visible = isJwtFeatureEnabled(state, 'recording', visible);
+    visible = isModerator
+        ? isJwtFeatureEnabled(state, 'recording', recordingService?.enabled || localRecordingEnabled || dropboxEnabled)
```
Comment (damencho) [inline react/features/recording/functions.ts:176]: Same and here, the last param of isJwtFeatureEnabled is used only when there is no token. But those are general configs for the service and they should overwrite the decision from isJwtFeatureEnabled.

**Review by damencho — CHANGES_REQUESTED** [review]

File: react/features/recording/components/LiveStream/AbstractLiveStreamButton.ts (hunk 1 above)
Comment (saghul) [reply react/features/recording/components/LiveStream/AbstractLiveStreamButton.ts:135]: Good point!

Comment (horymury) [reply react/features/recording/components/LiveStream/AbstractLiveStreamButton.ts:135]: this was existing logic - if we want the config to trump the jwt feature, I can change.

Comment (damencho) [reply react/features/recording/components/LiveStream/AbstractLiveStreamButton.ts:135]: oh there are more params no just the last one

**Review by damencho — APPROVED** [review]

**Comment by shawnchin** [comment]
Apologies for butting in, but does this affect local recording too? And if so, is that intentional -- i.e. to hide rec button for non-moderators even when only local recording is enabled?

**Comment by saghul** [comment]
Good question. @horymury ?

**Pull request merged**